infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/runtime/CorfuLogReplicationRuntime.java

```diff
@@ -146,7 +152,7 @@ public class CorfuLogReplicationRuntime {
 
     @Getter
     private final LogReplicationSourceManager sourceManager;
-    private volatile Set<String> connectedNodes;
+    private final Set<String> connectedNodes;
```

Comment:
can you explain why the 'volatile' was removed?

Reply:
`final` here implicitly guarantee `volatile` for this field as it was only assigned once in the constructor (didn't escape the constructor)

Reply:
hmm.. not sure I understand.  volatile guarantees that writes to this variable are applied before a subsequent read.  It ensures that a stale value is not read by any concurrently running thread.  'final' has a different purpose.

Reply:
The `final` signature here make the field immutable and other threads could only read its value after constructor finished, so here `final` is the same or even stronger than volatile